msp/mspimpl.go

```diff
@@ -825,10 +827,14 @@ func (msp *bccspmsp) sanitizeCert(cert *x509.Certificate) (*x509.Certificate, er
 // In this MSP implementation, well formed means that the PEM has a Type which is either
 // the string 'CERTIFICATE' or the Type is missing altogether.
 func (msp *bccspmsp) IsWellFormed(identity *m.SerializedIdentity) error {
-	bl, _ := pem.Decode(identity.IdBytes)
+	bl, rest := pem.Decode(identity.IdBytes)
```

Comment:
Thanks for the investigation, Yacov. pem.Decode is also used in deserializeIdentityInternal, isn't there a risk that there could be principal multiplication by adding trailing bytes there too?

Reply:
That's a good question, but I think there is no risk, because the MSP discards the PEM (and the DER) completely and just propagates the pure x509 certificate into the identity instance construction. 

Then, the identity identifier is computed over the raw DER of the certificate which is enclosed in the first (and only!) PEM block so we should be safe.

The reason I am doing this for gossip, is while gossip uses the MSP to validate identities, it carries around the representations received from the network and the MSP never takes part in sanitizing them.
This also means that gossip can theoretically save in memory different representations of the same identity - the low and the high S.
However I am OK with that, because gossip supports having several identities mapped to the same endpoint(s), and there is no harm in having 2 different representations.

The reason for this change is to prevent an infinite amount of representations which can flood the memory.

Reply:
@adecaro I'd like to know what is your opinion.

Reply:
So, I tried this

https://pastebin.com/tqD0Jvfj

and it works, the identity is valid despite the trailing bytes. This isn't a huge problem, because the identifier of the certificate, which is used for cert dedup, is the hash of the part that the x.509 library digests, so the trailing bytes are just dropped. Still, it's possible that this is an issue elsewhere.. Your thoughts?

Reply:
This is what I said above, we ignore the trailer when it is time to make access control decisions. Who cares if the identity has trailing bytes if you shave them off of it and forget about them?

`Still, it's possible that this is an issue elsewhere.. Your thoughts?`

Well what about the certificate header? We ignore it as well don't we? And I remember that in the past we never used "CERTIFICATE" to serialize the header, but used a blank string. 
Look at the comment in [IsWellFormed](https://github.com/hyperledger/fabric/blob/master/msp/mspimpl.go#L832-L836), it talks exactly about this. 
If you want to make a change without a capability you need to make sure it doesn't run anyone over.

Reply:
RIght, I guess a deeper analysis of all places where pem decoding is invoked might be warranted. I've seen it uses in the raft code as well for instance. Perhaps you could create a jira so we don't forget? Thanks

Reply:
OK I opened FAB-17811.